phc/web/patient_filter_query_builder.py

```diff
@@ -25,38 +25,77 @@ def __getattr__(self, attr):
+        if args and len(args) > 0:
+            raise NotImplementedError('Property only support key value arguments!')
+
+        components = []
+        for operator, value_or_values in kwargs.items():
+            values = value_or_values if type(value_or_values) is list else [value_or_values]
+            for value in values:
+                components.append({
+                    'operator': operator,
+                    'value': value
+                })
+
+        self.component_body[self.__global_handler.__func__.func_name] = components
         return self
 
     def to_dict(self):
         return self.component_body
 
 
-class PatientFilterComponentsQueryBuilder(BaseQueryBuilder):
+class ObservationProperty(Property):
+    def __init__(self):
+        Property.__init__(self)
+
+    def with_components(self, components_properties):
+        """
+        Handle component property.
+        """
+        if type(components_properties) is not list:
+            raise TypeError('Observation property with_components parameter must be in an array!')
+
+        self.component_body['components'] = components_properties
+        return self
+
+    def to_dict(self):
+        component_body_dict = dict()
+        for key, value_or_values in self.component_body.items():
+            if key == 'components':
+                if type(value_or_values) is list:
+                    component_values = []
+                    for value in value_or_values:
+                        component_values.append(value.to_dict())
+                    component_body_dict['components'] = component_values
+                else:
+                    component_body_dict['components'] = value_or_values.to_dict()
+            else:
+                component_body_dict[key] = value_or_values
+        return component_body_dict
+
+
+class Resource(BaseQueryBuilder):
     """
-    Property level builder class. 'observations', 'procedures' are considered as properties.
+    Resource level builder class. 'observations', 'procedures' are considered as resources.
     """
 
     def __init__(self):
         self.key_components = dict()
 
-    def observations(self):
-        search_component = PatientFilterComponentQueryBuilder()
-        self.key_components['observations'] = search_component
-        return search_component
+    def with_observations(self, observation_properties):
+        if type(observation_properties) is not list:
+            raise TypeError('Resource with_observations parameter must be in an array!')
+
+        self.key_components['observations'] = observation_properties
+        return self
 
     def to_dict(self):
         key_components_dict = dict()
         for key, components in self.key_components.items():
-            key_components_dict[key] = {**components.to_dict()}
+            components_dict = dict()
+            for component in components:
+                components_dict = {**components_dict, **component.to_dict()}
```

Comment:
It will likely be more efficient to use a nested comprehension here, avoiding creating a new object on each iteration, e.g. `components_dict = {key:value for x in components for key,value in x.items()}`